### PR TITLE
App metrics collapse | Build history page reload | Deployment history show loader

### PR DESCRIPTION
**Pull request by milindk8**
# Description

1. If application metrics not available then the card should be in a collapsed state
2. Build history: Page reloads when scrolling to bottom of build history list
3. Tab selection should retain the selection & No loader present when loading tab in deployment history

Fixes - https://github.com/devtron-labs/devtron/issues/2039, https://github.com/devtron-labs/devtron/issues/2040, https://github.com/devtron-labs/devtron/issues/2042

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 1]
```diff
@@ -353,39 +354,81 @@ function EnableAppMetrics() {
...
+                                <a
+                                    className="learn-more__href cta small text pl-0"
+                                    href={DOCUMENTATION.GLOBAL_CONFIG_CLUSTER}
+                                    target="_blank"
+                                    style={{ paddingLeft: '0' }}
+                                >
+                                    See how to fix
+                                </a>
+                                <Link
+                                    to={`${URLS.GLOBAL_CONFIG_CLUSTER}`}
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/AppMetrics.tsx:421]: we can directly write this like to={URLS.GLOBAL_CONFIG_CLUSTER}

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 2]
```diff
@@ -353,39 +354,81 @@ function EnableAppMetrics() {
...
+                            Unable to show metrics due to insufficient/incorrect configurations
+                        </p>
+                        {(!hostURLConfig || hostURLConfig.value !== window.location.origin) && (
+                            <>
+                                <p className="fw-4 fs-12 cn-7 mt-16 mb-0">
+                                    Host url is not configured or is incorrect. Reach out to your DevOps team
+                                    (super-admin) to configure host url.
+                                </p>
+                                <Link
+                                    to={`${URLS.GLOBAL_CONFIG_HOST_URL}`}
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/AppMetrics.tsx:401]: we can directly write this like to={URLS. GLOBAL_CONFIG_HOST_URL}

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 3]
```diff
@@ -353,39 +354,81 @@ function EnableAppMetrics() {
...
             </div>
-        </article>
-    </div>
+            {!collapsed && ( isLoading ? (
+                <div style={{ height: '240px' }}>
+                    <Progressing pageLoader />
+                </div>
+            ) : (
+                <article className="app-metrics-graph__empty-state">
+                    <img src={HostErrorImage} alt="" className="w-100" />
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/AppMetrics.tsx:389]: please put some value in alt

File: src/components/app/details/appDetails/AppMetrics.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppMetrics.tsx]: Done

File: src/components/app/details/appDetails/AppMetrics.tsx (hunk 3 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppMetrics.tsx]: done

File: src/components/app/details/appDetails/AppMetrics.tsx (hunk 2 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppMetrics.tsx]: done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**